test/SlowTests/Issues/RavenDB_21946.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using FastTests;
+using Raven.Client.Documents.Operations.Configuration;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Operations;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_21946 : RavenTestBase
+    {
+        public RavenDB_21946(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [RavenFact(RavenTestCategory.Configuration)]
+        public async Task ShouldUpdateLastModifiedTimeAfterConflictResolutionSave()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var collectionByScript = new Dictionary<string, ScriptResolver>
+                {
+                    {
+                        "Users", new ScriptResolver
+                       {
+                            Script = "Script1"
+                        }
+                    }
+                };
+
+                // configure conflict resolver with an initial script for the specified collection
+                await store.Maintenance.Server.SendAsync(new ModifyConflictSolverOperation(store.Database, collectionByScript, resolveToLatest: false));
+
+                var conflictSolver = await store.Maintenance.SendAsync(new GetConflictSolverConfigurationOperation());
+
+                Assert.False(conflictSolver.IsEmpty());
+                Assert.False(conflictSolver.ResolveToLatest);
+                Assert.Single(conflictSolver.ResolveByCollection);
+                Assert.True(conflictSolver.ResolveByCollection.TryGetValue("Users", out var scriptResolver1));
+
+                // wait to confirm LastModifiedTime remains unchanged over time
+                await Task.Delay(1000);
```

Comment:
use WaitForValueAsync instead

Reply:
I believe using `WaitForValueAsync` might miss the intended purpose here.. the goal of this delay is to confirm that the time remains unchanged over a specific period, whereas `WaitForValueAsync` is designed to return the expected result as soon as it’s available